Fails to build on older rust versions

<!--
─────────────────────────────────────────────
  ⚠️ IMPORTANT: Please run the following command to create an issue:

  starship bug-report

  An issue will be pre-populated with your system's configuration,
  making the process a whole lot quicker 😊
─────────────────────────────────────────────
-->

## Bug Report

#### Current Behavior
Cargo tries to build starship, but fails:

```
   Compiling starship v1.4.0
error[E0658]: use of unstable library feature 'available_parallelism'
 --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/starship-1.4.0/src/main.rs:5:5
  |
5 | use std::thread::available_parallelism;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #74479 <https://github.com/rust-lang/rust/issues/74479> for more information

error[E0658]: use of unstable library feature 'available_parallelism'
   --> /home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/starship-1.4.0/src/main.rs:235:28
    |
235 |         .unwrap_or_else(|| available_parallelism().map(usize::from).unwrap_or(1).min(8));
    |                            ^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #74479 <https://github.com/rust-lang/rust/issues/74479> for more information

For more information about this error, try `rustc --explain E0658`.
error: failed to compile `starship v1.4.0`, intermediate artifacts can be found at `/home/user/.cargo/target`

Caused by:
  could not compile `starship` due to 2 previous errors
```

#### Expected Behavior
It either should build or reasonable older versions, or `Cargo.toml` should specify a minimal rust version (see https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field).

#### Possible Solution
The used function `std::thread::available_parallelism` was added in rust's std version 1.59. See: https://doc.rust-lang.org/std/thread/fn.available_parallelism.html. Set `rust-version` to "1.59" in `Cargo.toml`.

#### Environment
Using rust installed via rustup.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -21,6 +21,9 @@ keywords = ["prompt", "shell", "bash", "fish", "zsh"]
 license = "ISC"
 readme = "README.md"
 repository = "https://github.com/starship/starship"
+# MSRV is specified to use std::thread::availabe_parallelism, which was stablizized in Rust version 1.59
+# Note: MSRV is only intended as a hint, and only the latest version is officially supported in starship.
+rust-version = "1.59"
 description = """
 The minimal, blazing-fast, and infinitely customizable prompt for any shell! ☄🌌️
 """

